packages/store/addon/-private/system/references/belongs-to.js

```diff
@@ -128,10 +130,17 @@ export default class BelongsToReference extends Reference {
     return resolve(objectOrPromise).then(data => {
       let record;
 
-      // TODO deprecate data as Model
-      if (peekRecordIdentifier(data)) {
-        record = data;
-      } else {
+      if (DEPRECATE_BELONGS_TO_REFERENCE_PUSH) {
+        if (peekRecordIdentifier(data)) {
```

Comment:
We can combine these to one line and keep the if/else branching to avoid the `record === undefined` check.

e.g.

```ts
if (DEPRECATE_BELONGS_TO_REFERENCE_PUSH && peekRecordIdentifier(data)) {

} else {

}
```